pkg/client/node_client_test.go

```diff
@@ -0,0 +1,638 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+
+	"sigs.k8s.io/container-object-storage-interface-api/apis/objectstorage.k8s.io/v1alpha1"
+	cosifake "sigs.k8s.io/container-object-storage-interface-api/clientset/fake"
+	cs "sigs.k8s.io/container-object-storage-interface-api/clientset/typed/objectstorage.k8s.io/v1alpha1"
+
+	"sigs.k8s.io/container-object-storage-interface-csi-adapter/pkg/util"
+	"sigs.k8s.io/container-object-storage-interface-csi-adapter/pkg/util/test"
+)
+
+var (
+	ctx = context.Background()
+)
+
+func TestGetBAR(t *testing.T) {
```

Comment:
Can you also add tests (or a TODO if it's a lot of effort) for checking the contents of the created file in the mountpoint given by NodeStageVolume 

Basically, we need tests for NodeStageVolume, NodePublishVolume, NodeUnstageVolume, NodeUnpublishVolume

Reply:
Would this be in the form of e2e tests? Currently, we aren't modifying anything on the host machine, and I am not sure if validating the contents of files/mounts would be a good fit for unit tests. 

As an aside, we are not using NodeStageVolume/NodeUnstageVolume. The only two (node server) implemented functions are NodePublishVolume/NodeUnpublishVolume